src/worker/tasks/wallet-subscription-worker.ts

```
@@ -0,0 +1,168 @@
+            await SendWebhookQueue.enqueueWebhook({
+              type: WebhooksEventTypes.WALLET_SUBSCRIPTION,
+              webhook: subscription.webhook,
+              body: {
+                subscriptionId: subscription.id,
+                chainId: subscription.chainId,
+                walletAddress: subscription.walletAddress,
+                condition,
+                currentValue,
+              },
+            });
+          }
+        }
+      } catch (error) {
+        // Log error but continue processing other subscriptions
+        const message = error instanceof Error ? error.message : String(error);
+        logger({
+          service: "worker",
+          level: "error",
+          message: `Error processing wallet subscription ${subscription.id}: ${message}`,
+          error: error as Error,
+        });
+      }
+    }),
+  );
+}
+
+// Must be explicitly called for the worker to run on this host.
+export const initWalletSubscriptionWorker = async () => {
+  const config = await getConfig();
+  const cronPattern = config.walletSubscriptionsCronSchedule || "*/30 * * * * *"; // Default to every 30 seconds
+
+  logger({
+    service: "worker",
+    level: "info",
+    message: `Initializing wallet subscription worker with cron pattern: ${cronPattern}`,
+  });
+
+  WalletSubscriptionQueue.q.add("cron", "", {
+    repeat: { pattern: cronPattern },
+    jobId: "wallet-subscription-cron",
+  });
+
+  const _worker = new Worker(WalletSubscriptionQueue.q.name, handler, {
+    connection: redis,
+    concurrency: 1,
+  });
+  logWorkerExceptions(_worker);
+};
+
+/**
+ * Process all wallet subscriptions and notify webhooks when conditions are met.
+ */
+const handler: Processor<string, void, string> = async (_job: Job<string>) => {
+  // Get all active wallet subscriptions
+  const subscriptions = await getAllWalletSubscriptions({
+    page: 1,
+    limit: 1000, // Process 1000 subscriptions at a time
+  });
```

Don't put a limit here because Engine allows creating > 1k subscriptions, and any more are silently ignored.

Instead just enforce this limit on the "create wallet subscription" endpoint: If > 1k non-deleted subscriptions, don't allow creating any more.